Autocoders/Python/schema/default/active_comp_schematron.rng

```diff
@@ -3,7 +3,7 @@
     xmlns="http://purl.oclc.org/dsdl/schematron">
 <pattern id = "ASYNC_PORT">
     <rule context="component">
-        <assert test="(//component/@kind = 'active' and count(//component/ports/port/@kind[. = 'async_input']) >= 1) or not (//component/@kind = 'active')">Active components should have at least 1 port of kind async_input.</assert>
+        <assert test="(//component/@kind = 'active' and (count(//component/ports/port/@kind[. = 'async_input']) + count(//component/internal_interfaces/internal_interface) ) >= 1) or not (//component/@kind = 'active')">Active components should have at least 1 port of kind async_input or internal_interface.</assert>
```

Comment:
I don't know too much about internal interfaces, but do they function as synchronous or async port calls? If they're async and require an active component I agree this is the correct fix.

Reply:
Internal interfaces don't require an active component. The use the message queue, so the component can be active or queued.

Reply:
Active components must have either an async port or an internal interface. This check is to make sure the Active component is well defined.  It will not affect queued component definitions.